test/EventParser.hs

```diff
@@ -0,0 +1,136 @@
+module EventParser
+    (
+      main
+    )
+where
+
+import Control.Monad (when)
+import Data.IORef (IORef, newIORef, modifyIORef, readIORef, writeIORef)
+import Data.Either (fromRight)
+import Data.Word
+import GHC.RTS.Events
+import Streamly.Internal.Serialize.FromBytes
+import Streamly.Data.Stream.Prelude
+import Streamly.Data.Parser
+
+import qualified Data.IntMap as Map
+import qualified Streamly.Data.Fold as Fold
+import qualified Streamly.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Parser.ParserD as PD
+import qualified Streamly.Internal.FileSystem.File as FP
+
+#define EVENT_HEADER_BEGIN    0x68647262
+#define EVENT_HEADER_END      0x68647265
+#define EVENT_DATA_BEGIN      0x64617462
+#define EVENT_DATA_END        0xffff
+#define EVENT_HET_BEGIN       0x68657462
+#define EVENT_HET_END         0x68657465
+#define EVENT_ET_BEGIN        0x65746200
+#define EVENT_ET_END          0x65746500
+
+parseF :: IORef (Map.IntMap Word16) -> FilePath -> IO ()
+parseF kvRef path = do
+    (res, nxt) <- Stream.parseBreak word32be $ FP.read path
```

Comment:
StreamD.parseBreak should not be used, it will give you quadratic perf. We should use StreamK.parseBreak.